api/search.go

```diff
@@ -208,13 +219,90 @@ func CreateRequests(w http.ResponseWriter, req *http.Request, validator QueryPar
 	return q, reqSearch, reqCount
 }
 
+func NLPSearchHandlerFunc(cli *nlp.Client) http.HandlerFunc {
+	return func(w http.ResponseWriter, req *http.Request) {
+		ctx := req.Context()
+		params := req.URL.Query()
+
+		var berlin models.Berlin
+		var scrubber models.Scrubber
+		var category models.Category
+
+		var wg sync.WaitGroup
+		wg.Add(2)
+
+		scrubber, err := cli.GetScrubber(ctx, params.Get("q"))
+		if err != nil {
+			log.Error(ctx, "error making request to scrubber: %w", err)
+		}
+
+		go func() {
+			defer wg.Done()
+
+			var err error
+
+			category, err = cli.GetCategory(ctx, scrubber.Query)
+			if err != nil {
+				log.Error(ctx, "error making request to category: %w", err)
+			}
+		}()
+
+		go func() {
+			defer wg.Done()
+
+			var err error
+
+			berlin, err = cli.GetBerlin(ctx, scrubber.Query)
+			if err != nil {
+				log.Error(ctx, "error making request to berlin: %w", err)
+			}
+		}()
+
+		wg.Wait()
+
+		resp := models.NLPResp{
+			Berlin:   berlin,
+			Scrubber: scrubber,
+			Category: category,
+		}
+
+		responseSearchData, err := json.Marshal(resp)
+		if err != nil {
+			log.Error(ctx, "marshaling response data failed", err)
+		}
+
+		w.Header().Set("Content-Type", "application/json;charset=utf-8")
+		if _, err := w.Write(responseSearchData); err != nil {
+			log.Error(ctx, "writing response failed", err)
+			http.Error(w, "Failed to write http response", http.StatusInternalServerError)
+			return
+		}
+	}
+}
+
 // SearchHandlerFunc returns a http handler function handling search api requests.
-func SearchHandlerFunc(validator QueryParamValidator, queryBuilder QueryBuilder, elasticSearchClient DpElasticSearcher, transformer ResponseTransformer) http.HandlerFunc {
+func SearchHandlerFunc(validator QueryParamValidator, queryBuilder QueryBuilder, nlpConfig config.NLP, nlpCLI *nlp.Client, elasticSearchClient DpElasticSearcher, transformer ResponseTransformer) http.HandlerFunc {
 	return func(w http.ResponseWriter, req *http.Request) {
 		ctx := req.Context()
 		params := req.URL.Query()
 
-		q, searchReq, countReq := CreateRequests(w, req, validator)
+		q := params.Get("q")
+
+		var nlpCriteria *query.NlpCriteria
+		if params.Get("c") == "1" {
```

Comment:
Is this idiomatic within dp-search-api or is there any other preferred way for testing a boolean parameter is on? (e.g. is there a comparable example elsewhere?)